core/src/migrations/000075-polymorphicFilters.ts

```diff
@@ -0,0 +1,45 @@
+export default {
+  up: async function (migration, DataTypes) {
+    await migration.sequelize.transaction(async () => {
+      await migration.renameTable("propertyFilters", "filters");
+
+      await migration.renameColumn("filters", "propertyId", "ownerId");
+
+      await migration.addColumn("filters", "ownerType", {
+        type: DataTypes.STRING(191),
+        allowNull: false,
+        defaultValue: "property",
+      });
+
+      await migration.changeColumn("filters", "ownerType", {
+        type: DataTypes.STRING(191),
+        allowNull: false,
+      });
+
+      await migration.addIndex(
+        "filters",
+        ["ownerId", "ownerType", "position"],
+        {
+          fields: ["ownerId", "ownerType", "position"],
+          unique: true,
+        }
+      );
```

Comment:
You can use the same column multiple filters (userId > 0 and userId < 100), so the only real validation we can do in the database is that there's one entry for each position per parent.